wgpu-hal/src/dx12/adapter.rs

```diff
@@ -672,16 +672,20 @@ impl crate::Adapter for super::Adapter {
         );
         // UAVs use srv_uav_format
         caps.set(
-            Tfc::STORAGE_WRITE_ONLY,
+            Tfc::STORAGE_READ_ONLY,
             data_srv_uav
-                .Support1
-                .contains(Direct3D12::D3D12_FORMAT_SUPPORT1_TYPED_UNORDERED_ACCESS_VIEW),
```

Comment:
note: My understanding is that we're these checks from:

* `D3D12_FORMAT_SUPPORT1_TYPED_UNORDERED_ACCESS_VIEW` allows `STORAGE_READ_ONLY`.
* `D3D12_FORMAT_SUPPORT2_UAV_TYPED_LOAD` allows `STORAGE_{READ_ONLY,WRITE_ONLY,READ_WRITE}`.

…to:

* `D3D12_FORMAT_SUPPORT2_UAV_TYPED_LOAD` allows `STORAGE_READ_ONLY`.
* `D3D12_FORMAT_SUPPORT2_UAV_TYPED_STORE` allows `STORAGE_WRITE_ONLY`.
* Allowing both `STORAGE_READ_ONLY` and `STORAGE_WRITE_ONLY` allows `STORAGE_READ_WRITE`.

Reply:
question: AIUI, `D3D12_FORMAT_SUPPORT1_TYPED_UNORDERED_ACCESS_VIEW` implies _at least one_ of `D3D12_FORMAT_SUPPORT2_UAV_TYPED_LOAD` or `D3D12_FORMAT_SUPPORT2_UAV_TYPED_STORE`, but that's it. It seems like allowing _any_ of these capabilities based on that was wrong because...it's too broad. Which would mean the `STORAGE_WRITE_ONLY` was previously being allowed in too many cases. Is this thinking right?

Reply:
That's my understanding as well.